uwblueprint/supportive-housing
# Cascading Delete Tags

## Notion task link


[Task Name](https://www.notion.so/uwblueprintexecs/Cascading-Tags-58fb5fab828147748139026c4a4d81dc?pvs=4)



## Implementation description
* Implemented the cascading delete so that it deletes all log record references with the associated tag and replaces that status field in the tags database



## Steps to test
1. Migrate your database and upgrade
2. Create sample tags in your tags database. Ex: INSERT into tags (name, status) VALUES ('tagA', 'Active');
3. Then, in the browser, create log records that have tags associated with them. Example body: 
`{
  "employee_id": 7,
  "resident_id": 2,
  "datetime": "2023-11-12T01:56:01.891Z",
  "flagged": false,
  "note": "lalala",
  "tags": ["tagA", "tagB", "tagC"],
  "building_id": 1
}
`
4. Then, try deleting a tag that is associated with the log record. Verify in your log_record_tag junction table that all entries with that tag_id have now been deleted.
5. Verify that the status column no longer exists in your database




## What should reviewers focus on?
* 


## Checklist
